### PR TITLE
Nix CI: run integration tests

### PR DESCRIPTION
**Pull request by balsoft**
Run integration tests from the Nix CI

Closes https://github.com/MinaProtocol/mina/issues/11126

### CODE REVIEW

**Comment by balsoft** [comment]
Closes #11126

**Comment by robinbb** [comment]
@balsoft It appears that CI is red. Likely, no reviewer wants to review a broken branch. Please mark this as 'draft' if you are not working on it. Please make the CI green if you are working on it and want a review; please then obtain the review.

/cc @mzajda0mina

**Comment by robinbb** [comment]
@balsoft Merging something into a branch that is not 'develop' does not close the associated issue, FYI.

**Comment by balsoft** [comment]
> It appears that CI is red
Yes, sorry, my mistake. I pushed a broken merge commit.

**Comment by balsoft** [comment]
Both the failing integration tests (opt-block-prod and peers-reliability) succeed when I trigger them locally. This is really weird.

**Comment by kaozenn** [comment]
Integrations assumes that the artifacts are deployed on the specific cluster: `gke_o1labs-192920_us-west1_mina-integration-west1` on GCP before running the test executive. I understand this aims to map the current workflow. However, it creates a strong coupling between the code base and the cloud provider.

**Comment by balsoft** [comment]
!ci-build-me

**Comment by balsoft** [comment]
!ci-build-me

**Comment by balsoft** [comment]
!ci-build-me

File: flake.nix [hunk 1]
```diff
@@ -187,6 +188,39 @@
...
+          runIntegrationTest = test:
+            { with-archive ? false }: {
+              command =
+                runInEnv self.devShells.x86_64-linux.integration-tests ''
+                  export GOOGLE_CLOUD_KEYFILE_JSON=$AUTOMATED_VALIDATION_SERVICE_ACCOUNT
+                  export GCLOUD_API_KEY=$(cat $INTEGRATION_TEST_LOGS_GCLOUD_API_KEY_PATH)
+                  source $INTEGRATION_TEST_CREDENTIALS
+                  export AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY
+                  export KUBE_CONFIG_PATH=$$HOME/.kube/config
+                  gcloud auth activate-service-account --key-file=$AUTOMATED_VALIDATION_SERVICE_ACCOUNT automated-validation@o1labs-192920.iam.gserviceaccount.com --project o1labs-192920
```
Comment (bryanhonof) [inline flake.nix:200]: Won't these variables end up in the Nix store? If yes, that'd be a bad security posture. Perhaps even related to #10692.

**Review by bryanhonof — APPROVED** [review]
Apart from the one comment, looks good to me.

File: flake.nix (hunk 1 above)
Comment (balsoft) [reply flake.nix:200]: Nope, this is performed completely outside the store. What ends up in the pipeline is only the variable names, the contents are read at "runtime" during step execution.

Comment (bryanhonof) [reply flake.nix:200]: Ah, right. Makes sense!

**Pull request merged**